ethereum/core/src/main/java/org/hyperledger/besu/ethereum/ProtocolContext.java

```diff
@@ -67,4 +68,14 @@ public <C extends ConsensusContext> Optional<C> safeConsensusContext(final Class
         .filter(c -> klass.isAssignableFrom(c.getClass()))
         .map(klass::cast);
   }
+
+  // TODO: cleanup - isChainPruningEnabled will not be required after
+  // https://github.com/hyperledger/besu/pull/4703 is merged.
+  public void setIsChainPruningEnabled(final boolean isChainPruningEnabled) {
```

Comment:
Don't think this is a good place for the chain pruning enabled flag, it is very high level. Maybe part of the Merge context would be ok, as we only need this check for the Merge RPCs.

Reply:
Updated. Ended up placing a flag in merge context since it involves very little change.